src/explosion.cpp

```diff
@@ -655,36 +655,34 @@ void scrambler_blast( const tripoint &p )
 
 void emp_blast( const tripoint &p )
 {
-    // TODO: Implement z part
-    point p2( p.xy() );
     Character &player_character = get_player_character();
     const bool sight = player_character.sees( p );
     map &here = get_map();
-    if( here.has_flag( "CONSOLE", p2 ) ) {
+    if( here.has_flag( "CONSOLE", p ) ) {
         if( sight ) {
-            add_msg( _( "The %s is rendered non-functional!" ), here.tername( p2 ) );
+            add_msg( _( "The %s is rendered non-functional!" ), here.tername( p ) );
         }
-        here.furn_set( p2, furn_str_id( "f_machinery_electronic" ) );
+        here.furn_set( p, furn_str_id( "f_machinery_electronic" ) );
         return;
     }
     // TODO: More terrain effects.
-    if( here.ter( p2 ) == t_card_science || here.ter( p2 ) == t_card_military ||
-        here.ter( p2 ) == t_card_industrial ) {
+    if( here.ter( p ) == t_card_science || here.ter( p ) == t_card_military ||
+        here.ter( p ) == t_card_industrial ) {
         int rn = rng( 1, 100 );
         if( rn > 92 || rn < 40 ) {
             if( sight ) {
                 add_msg( _( "The card reader is rendered non-functional." ) );
             }
-            here.ter_set( p2, t_card_reader_broken );
+            here.ter_set( p, t_card_reader_broken );
         }
         if( rn > 80 ) {
             if( sight ) {
                 add_msg( _( "The nearby doors slide open!" ) );
             }
             for( int i = -3; i <= 3; i++ ) {
                 for( int j = -3; j <= 3; j++ ) {
-                    if( here.ter( p2 + point( i, j ) ) == t_door_metal_locked ) {
-                        here.ter_set( p2 + point( i, j ), t_floor );
+                    if( here.ter( p + tripoint( i, j, 0 ) ) == t_door_metal_locked ) {
+                        here.ter_set( p + tripoint( i, j, 0 ), t_floor );
```

Comment:
For your future reference: you didn't need to change the type of the right-hand addend here; adding a `tripoint` to a `point` works fine.